x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/TopNOperator.java

```diff
@@ -194,6 +198,16 @@ private void writeValues(int position, BreakingBytesRefBuilder values) {
         }
     }
 
+    public record Partition(int channel) {
+
+        private static final long SHALLOW_SIZE = RamUsageEstimator.shallowSizeOfInstance(Partition.class);
```

Comment:
Do we use the `SHALLOW_SIZE`?

Reply:
I used it in `ramBytesUsed` method (https://github.com/elastic/elasticsearch/pull/128135/files#diff-7084c863fd94b64144b18c7ab2f8fb7958e8288d4e6aca6fad876ec39e0b06acR602) similarily to how it's used for sort orders.